### DISCUSSION TITLE
external-ui-url 参数要如何生效？

### DISCUSSION THREAD

**Discussion by KinShin** [MetaCubeX/mihomo#1131] [Q&A] [answered]
看到配置文件中有 ‎external-ui-name ‎external-ui-url 参数，看着像是从远程下载和更新 webui 的功能。

我参照文档配置好 external-controller external-ui，并确保路径正确、端口无冲突等情况下，发现 mihomo 并没有将 external-ui-url 的文件下载并解压到 external-ui 的目录下，访问 http://{{external-controller}}/ui 也只有空白页。

是我对这个功能理解有误还是目前的版本还没有实现获取远程 webui 的功能？

**Comment by Victor2333** [comment, marked as answer]
看了代码 是暴露了一个接口去做下载的
POST /upgrade/ui  body为空就可以了
secret 放header里

**Reply by KinShin** [reply]
感谢🙏🏻，原来是给 webui 用的。

**Reply by Victor2333** [reply]
我也因为这个参数迷茫了好久，只是之前在看tun fakeip在docker不工作的原因，没仔细研究，正好今天看了下
